### PR TITLE
swagger integration

### PR DESCRIPTION
**Pull request by dinesh-aot**
- Common Decorator added
- Swagger doc annotations added
image

### CODE REVIEW

File: scaffold-api/src/scaffold_api/resources/apihelper.py [hunk 1]
```diff
@@ -26,10 +31,70 @@ class Api(BaseApi):
...
+            @api.response(500, "Internal Server Error")
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            return wrapper
+
+        return decorator
+
+    @classmethod
+    def convert_ma_schema_to_restx_model(cls, api, schema, name):
```
Comment (jadmnsaad) [inline scaffold-api/src/scaffold_api/resources/apihelper.py:51]: from Gemini:

from flask_restx import fields, Api

def convert_ma_schema_to_restx_model(api: Api, schema: ma.Schema, name: str) -> fields.Model:
    """Converts a Marshmallow schema to a Flask-RESTX model, handling nested fields.

    Args:
        api: The Flask-RESTX API instance.
        schema: The Marshmallow schema instance.
        name: The desired name for the Flask-RESTX model.

    Returns:
        A Flask-RESTX model with nested fields support.

    Raises:
        ValueError: If an unsupported Marshmallow field type is encountered.
    """

    model_fields = {}
    for field_name, field in schema.fields.items():
        field_type = type(field)

        # Basic field types
        if field_type in (ma_fields.Integer, ma_fields.String, ma_fields.Float,
                          ma_fields.Boolean, ma_fields.DateTime):
            model_fields[field_name] = _get_basic_field(field)

        # Nested fields
        elif field_type == ma.fields.Nested:
            nested_schema = field.schema
            model_fields[field_name] = fields.Nested(
                convert_ma_schema_to_restx_model(api, nested_schema, f"{name}_{field_name}")
            )

        else:
            raise ValueError(f"Unsupported Marshmallow field type: {field_type}")

    return api.model(name, model_fields)

def _get_basic_field(field: ma.fields.Field) -> fields.Field:
    """Creates a basic Flask-RESTX field from a Marshmallow field.

    Args:
        field: The Marshmallow field.

    Returns:
        A corresponding Flask-RESTX field.
    """

    return fields.Field(
        required=field.required,
        description=field.metadata.get("description", ""),
    )

Comment (jadmnsaad) [reply scaffold-api/src/scaffold_api/resources/apihelper.py:51]: To just make it more compact and to consider nested fields

File: scaffold-api/src/scaffold_api/resources/user.py [hunk 2]
```diff
@@ -15,69 +15,88 @@
...
-    @auth.require
+    @ApiHelper.common_decorator(API, endpoint_description="Fetch a user by id")
+    @API.response(code=200, model=user_list_model, description="Success")
+    @API.response(404, "Not Found")
     def get(user_id):
         """Fetch a user by id."""
         user = UserService.get_user_by_id(user_id)
-        return user, HTTPStatus.OK
+        if not user:
+            return UserSchema().dump(user), HTTPStatus.NOT_FOUND
```
Comment (jadmnsaad) [inline scaffold-api/src/scaffold_api/resources/user.py]: if not user we still dump it? It's null safe then right?

File: scaffold-api/src/scaffold_api/resources/apihelper.py [hunk 3]
```diff
@@ -26,10 +31,70 @@ class Api(BaseApi):
...
+        return url_for(self.endpoint("specs"), _external=True, _scheme=scheme)
+
+    @classmethod
+    def common_decorator(cls, api, endpoint_description):
+        """Common decorators for the resources"""
+
+        def decorator(func):
+            @wraps(func)
+            @cross_origin(origins=allowedorigins())
+            @auth.require
```
Comment (jadmnsaad) [inline scaffold-api/src/scaffold_api/resources/apihelper.py]: maybe we have the auth things in another common decorator? they are optional and I remember in met we had things like auth.require auth.optional and auth.has_one_of_roles. Our auth needs might become more complex and might not fit in a common decorator

File: scaffold-api/src/scaffold_api/resources/user.py (hunk 2 above)
Comment (dinesh-aot) [reply scaffold-api/src/scaffold_api/resources/user.py]: I will use the custom exceptions for this

**Review by nitheesh-aot — APPROVED** [review]

**Pull request merged**